### PR TITLE
[FR] Make modelname as a default parameter

### PR DESCRIPTION
**Pull request by maririos**
From arch board feedback, moving `ModelName` out of options bag for training and composedModel endpoints and add it as a default parameter.
documentation and tests updated

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs [hunk 1]
```diff
@@ -216,9 +300,9 @@ public virtual CreateComposedModelOperation StartCreateComposedModel(IEnumerable
                 }
 
                 var composeRequest = new ComposeRequest(modelIdsGuid);
-                if (createComposedModelOptions.ModelName.Length > 0)
+                if (!string.IsNullOrEmpty(modelName))
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs:303]: just curious, why do we do client validation on the modelName here, but not in StartTraining?

**Review by kristapratico — APPROVED** [review]
LGTM

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs (hunk 1 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs:303]: Hmm great point. I was using previous logic which made more sense but now it is extra. I will remove

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs]: Thank you!
Updated

**Review by kinelski — APPROVED** [review]
LGTM!

**Pull request merged**